src/__tests__/extensions/replay/sessionrecording.test.ts

```diff
@@ -2312,48 +2311,55 @@ describe('SessionRecording', () => {
             )
         })
 
-        it('flushes buffer and includes pause event when hitting blocked URL', async () => {
+        it('does not flush buffer and includes pause event when hitting blocked URL', async () => {
```

Comment:
that means that flushing before pausing is no longer safe.

really we should detect if we've ever flushed and then flush or not based on that but it's better to avoid this completely